idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,361 @@
+            return b.volume_bed == (constants.pi * b.bed_height *
+                                    (0.5 * b.bed_diameter) ** 2
+                                    )
+
+        # Solid phase material balance
+        # Volume of solid
+        self.volume_solid = Var(
+                    self.flowsheet().config.time,
+                    initialize=1.0,
+                    doc="Solids phase volume including particle pores  [m**3]")
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Calculating solid phase volume")
+        def volume_solid_constraint(b, t):
+            return b.volume_solid[t] == (b.volume_bed *
+                                         (1 - b.solids[t]._params.voidage)
+                                         )
+
+        # Accumulation equal to mass transfer/reaction
+        self.solids_material_holdup = Var(
+                self.flowsheet().time,
+                self.config.solid_property_package.component_list,
+                initialize=1,
+                doc="Solid phase component holdups",
+                units=pyunits.kg)
+
+        @self.Constraint(self.flowsheet().config.time,
+                         self.config.solid_property_package.component_list,
+                         doc="Solid phase material holdup constraints")
+        def solids_material_holdup_constraints(b, t, j):
+            return b.solids_material_holdup[t, j] == (
+                b.volume_solid[t] *
+                b.solids[t].get_material_density_terms("Sol", j))
+
+        self.solids_material_accumulation = DerivativeVar(
+                    self.solids_material_holdup,
+                    wrt=self.flowsheet().config.time,
+                    doc="Solids material accumulation",
+                    units=pyunits.kg/pyunits.s)
+
+        @self.Constraint(self.flowsheet().config.time,
+                         self.config.solid_property_package.component_list,
+                         doc="Solid phase material accumulation constraints")
+        def solids_material_accumulation_constraints(b, t, j):
+            return (b.solids_material_accumulation[t, j] ==
+                    b.volume_solid[t] *
+                    b.solids[t]._params.mw_comp[j] *
+                    sum(b.reactions[t].reaction_rate[r] *
+                        b.config.reaction_package
+                        .rate_reaction_stoichiometry[r, "Sol", j] for r in
+                        b.config.reaction_package.rate_reaction_idx))
+
+        # Add solid mass variable and constraint for TGA tracking
+        self.mass_solids = Var(self.flowsheet().config.time,
+                               doc="Total mass of solids",
+                               units=pyunits.kg)
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Calculating total mass of solids")
+        def mass_solids_constraint(b, t):
+            return 1e2*b.mass_solids[t] == 1e2*b.volume_solid[t] * sum(
```

Comment:
We really should be switching to use the IDAES scaling tools rather than hard coding scaling factors in our constraints.

Reply:
After importing the scaling tools as `sc`, would adding `sc.__scale_constraint(mass_solids_constraint(b, t), 100)` on line 240 be the correct way to do this?

Reply:
No - you should add a separate method named `calculate_scaling_factors` where you use the `constraint_scaling_transformation` method. Look at Line 1024 in https://github.com/IDAES/idaes-pse/blob/main/idaes/generic_models/unit_models/pressure_changer.py.